src/api/controllers/cache-controller.ts

```diff
@@ -48,31 +59,30 @@ export function setResponseNonCacheable(res: Response) {
 }
 
 /**
- * Sets the response `Cache-Control` and `ETag` headers using the chain tip previously added
+ * Sets the response `Cache-Control` and `ETag` headers using the etag previously added
  * to the response locals.
- * Uses the latest unanchored microblock hash if available, otherwise uses the anchor
- * block index hash.
  */
-export function setChainTipCacheHeaders(res: Response) {
-  const chainTip: FoundOrNot<DbChainTip> | undefined = res.locals[CHAIN_TIP_LOCAL];
-  if (!chainTip) {
-    logger.error(
-      `Cannot set cache control headers, no chain tip was set on \`Response.locals[CHAIN_TIP_LOCAL]\`.`
-    );
-    return;
-  }
-  if (!chainTip.found) {
+export function setETagCacheHeaders(res: Response, etagType: ETagType = ETagType.chainTip) {
+  const etag: string | undefined = res.locals[etagType];
+  if (!etag) {
+    // An empty mempool ETag means either no pending mempool txs or that the API is running
+    // on old pg versions. No error reporting is necessary in that case.
```

Comment:
I have mild concerns with maintainability if we get rid of this error condition / logging. It's pretty easy to incorrectly setup cache control on a given endpoint, and this is one of the areas that is able to detect that. 

Seems like it would be useful to retain that behavior. What do you think?

Reply:
Yeah that sounds like a good approach. The previous behavior was simply "check if this locals property was set, regardless if empty value", then error if the property is not set (since that would a bug / 500 error).